In [2]:
import pypsa
import pandas as pd

# Process and analysis

## Network without NEP

In [ ]:
n_no_nep = pypsa.Network("../results/test_scenario_wo_nep/prenetworks-brownfield/base_s_52_lv1.17___2035.nc")

In [4]:
ac_lines = n_no_nep.lines

In [5]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

135576.78876656335

In [6]:
dc_links = n_no_nep.links.query("carrier=='DC'")
dc_links.head()

bus0    bus1 type carrier  efficiency  active  build_year  \
Link                                                                            
relation/5487095   DK0 0   DK1 0           DC    0.975075    True           0   
relation/8193755   BE0 0  DE0 30           DC    0.976101    True           0   
relation/2505320   DK1 0  DE0 13           DC    0.974026    True           0   
relation/3918230  DE0 11   SE1 0           DC    0.955191    True           0   
relation/8185420   DK0 0   NL0 0           DC    0.966905    True           0   

                  lifetime   p_nom  p_nom_mod  ...  efficiency2  voltage  \
Link                                           ...                         
relation/5487095       inf   600.0        0.0  ...          1.0    400.0   
relation/8193755       inf  1000.0        0.0  ...          1.0    320.0   
relation/2505320       inf   600.0        0.0  ...          1.0    400.0   
relation/3918230       inf   600.0        0.0  ...          1.0    450.0   
relation/8185420       inf   700.0        0.0  ...          1.0    320.0   

                                                           geometry  \
Link                                                                  
relation/5487095  LINESTRING (10.5069 55.3673, 10.5068756 55.367...   
relation/8193755  LINESTRING (5.6675 50.753, 5.6675409 50.752914...   
relation/2505320  LINESTRING (12.0074 55.4499, 12.0073721 55.449...   
relation/3918230  LINESTRING (10.8022 53.8958, 10.802197 53.8958...   
relation/8185420  LINESTRING (8.7045 55.5305, 8.7030898 55.53044...   

                  underwater_fraction  length_original  underground  reversed  \
Link                                                                            
relation/5487095             0.563251       216.510855          1.0     False   
relation/8193755             0.000000       171.308786          1.0     False   
relation/2505320             0.281098       262.778550          1.0     False   
relation/3918230             0.927588      1101.945135          1.0     False   
relation/8185420             0.926393       578.122730          1.0     False   

                  under_construction  location   dc  
Link                                                 
relation/5487095                 0.0            1.0  
relation/8193755                 0.0            1.0  
relation/2505320                 0.0            1.0  
relation/3918230                 0.0            1.0  
relation/8185420                 0.0            1.0  

[5 rows x 49 columns]

In [7]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

3790.162021813663

In [8]:
grid_cap_wo_nep = line_cap + dc_links_cap
grid_cap_wo_nep

139366.950788377

## Network with NEP lines

In [9]:
n_nep = pypsa.Network("../results/test_scenario_w_nep_lines/prenetworks-brownfield/base_s_52_lv1.17___2035.nc")

INFO:pypsa.io:Imported network base_s_52_lv1.17___2035.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores


In [10]:
n_nep.statistics.installed_capacity()

component    carrier                                     
Link         BEV charger                                      621033.78150
             Combined-Cycle Gas                               164204.56410
             DC                                                12760.00000
             Open-Cycle Gas                                       89.90361
             V2G                                              621033.78150
             coal                                              27700.77775
             lignite                                           67353.58717
             nuclear                                           72996.93252
             oil                                                2658.70986
             oil refining                                    1000000.00000
             residential rural gas boiler                      91918.76897
             residential rural ground heat pump                 3392.85928
             residential urban decentral a

In [11]:
ac_lines = n_nep.lines

In [12]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

142307.6931630391

In [13]:
dc_links = n_nep.links.query("carrier=='DC'")
dc_links.head()

bus0    bus1 type carrier  efficiency  active  build_year  \
Link                                                                            
relation/5487095   DK0 0   DK1 0           DC    0.975075    True           0   
relation/8193755   BE0 0  DE0 24           DC    0.976280    True           0   
relation/2505320   DK1 0  DE0 11           DC    0.974026    True           0   
relation/3918230  DE0 25   SE1 0           DC    0.955169    True           0   
relation/8185420   DK0 0   NL0 0           DC    0.966905    True           0   

                  lifetime   p_nom  p_nom_mod  ...  underwater_fraction  \
Link                                           ...                        
relation/5487095       inf   600.0        0.0  ...             0.563251   
relation/8193755       inf  1000.0        0.0  ...             0.000000   
relation/2505320       inf   600.0        0.0  ...             0.281098   
relation/3918230       inf   600.0        0.0  ...             0.927588   
relation/8185420       inf   700.0        0.0  ...             0.926393   

                                                           geometry  \
Link                                                                  
relation/5487095  LINESTRING (10.5069 55.3673, 10.5068756 55.367...   
relation/8193755  LINESTRING (5.6675 50.753, 5.6675409 50.752914...   
relation/2505320  LINESTRING (12.0074 55.4499, 12.0073721 55.449...   
relation/3918230  LINESTRING (10.8022 53.8958, 10.802197 53.8958...   
relation/8185420  LINESTRING (8.7045 55.5305, 8.7030898 55.53044...   

                  length_original  underground  reversed  under_construction  \
Link                                                                           
relation/5487095       216.510855          1.0     False                 0.0   
relation/8193755       163.444681          1.0     False                 0.0   
relation/2505320       262.778550          1.0     False                 0.0   
relation/3918230      1102.948245          1.0     False                 0.0   
relation/8185420       578.122730          1.0     False                 0.0   

                  location  project_status  tags   dc  
Link                                                   
relation/5487095                                  1.0  
relation/8193755                                  1.0  
relation/2505320                                  1.0  
relation/3918230                                  1.0  
relation/8185420                                  1.0  

[5 rows x 51 columns]

In [14]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

3782.8997825830643

In [15]:
grid_cap_w_nep_lines = line_cap + dc_links_cap
grid_cap_w_nep_lines

146090.59294562216

## Network with NEP lines + links

In [16]:
n_nep_lines_links = pypsa.Network("../results/test_scenario_w_nep_lines_links/prenetworks-brownfield/base_s_52_lv1.17___2035.nc")

INFO:pypsa.io:Imported network base_s_52_lv1.17___2035.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores


In [17]:
n_nep_lines_links.statistics.installed_capacity()

component    carrier                                     
Link         BEV charger                                      621033.78150
             Combined-Cycle Gas                               164204.56410
             DC                                                80760.00000
             Open-Cycle Gas                                       89.90361
             V2G                                              621033.78150
             coal                                              27700.77775
             lignite                                           67353.58717
             nuclear                                           72996.93252
             oil                                                2658.70986
             oil refining                                    1000000.00000
             residential rural gas boiler                      91918.76897
             residential rural resistive heater                24886.20183
             residential urban decentral a

In [18]:
ac_lines = n_nep_lines_links.lines

In [19]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

142307.6931630391

In [20]:
dc_links = n_nep_lines_links.links.query("carrier=='DC'")
dc_links.head()

bus0    bus1 type carrier  efficiency  active  build_year  \
Link                                                                            
relation/5487095   DK0 0   DK1 0           DC    0.975075    True           0   
relation/8193755   BE0 0  DE0 24           DC    0.976280    True           0   
relation/2505320   DK1 0  DE0 11           DC    0.974026    True           0   
relation/3918230  DE0 25   SE1 0           DC    0.955169    True           0   
relation/8185420   DK0 0   NL0 0           DC    0.966905    True           0   

                  lifetime   p_nom  p_nom_mod  ...  underwater_fraction  \
Link                                           ...                        
relation/5487095       inf   600.0        0.0  ...             0.563251   
relation/8193755       inf  1000.0        0.0  ...             0.000000   
relation/2505320       inf   600.0        0.0  ...             0.281098   
relation/3918230       inf   600.0        0.0  ...             0.927588   
relation/8185420       inf   700.0        0.0  ...             0.926393   

                                                           geometry  reversed  \
Link                                                                            
relation/5487095  LINESTRING (10.5069 55.3673, 10.5068756 55.367...     False   
relation/8193755  LINESTRING (5.6675 50.753, 5.6675409 50.752914...     False   
relation/2505320  LINESTRING (12.0074 55.4499, 12.0073721 55.449...     False   
relation/3918230  LINESTRING (10.8022 53.8958, 10.802197 53.8958...     False   
relation/8185420  LINESTRING (8.7045 55.5305, 8.7030898 55.53044...     False   

                  underground  length_original  under_construction  location  \
Link                                                                           
relation/5487095          1.0       216.510855                 0.0             
relation/8193755          1.0       163.444681                 0.0             
relation/2505320          1.0       262.778550                 0.0             
relation/3918230          1.0      1102.948245                 0.0             
relation/8185420          1.0       578.122730                 0.0             

                  project_status  tags   dc  
Link                                         
relation/5487095                        1.0  
relation/8193755                        1.0  
relation/2505320                        1.0  
relation/3918230                        1.0  
relation/8185420                        1.0  

[5 rows x 51 columns]

In [21]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

20220.436007161625

In [22]:
grid_cap_w_nep_lines_links = line_cap + dc_links_cap
grid_cap_w_nep_lines_links

162528.12917020073

## Calculate ratios

In [23]:
# Ratio with NEP lines + links and without NEP
grid_cap_w_nep_lines_links/grid_cap_wo_nep

1.1661884560923848

In [24]:
# Ratio with NEP lines + links and NEP lines
grid_cap_w_nep_lines_links/grid_cap_w_nep_lines

1.1125160483857912

## Carbon emission targets

In [7]:
ctrys = ['AT', 'BE', 'CH', 'CZ', 'DE', 'DK', 'FR', 'IT', 'LU', 'NL', 'PL', 'SE']

In [8]:

# 1990 emissions in MtCO2
co2_totals = pd.read_csv("../data/co2_totals_1990.csv", index_col=0).loc[ctrys]

In [18]:
# 2005 emissions in MtCO2
co2_totals_2005 = pd.read_csv("../data/co2_totals_2005.csv", index_col=0).loc[ctrys]

In [19]:
# Total emissions excl. LULUCF
co2_totals_excl_lulucf = co2_totals.drop(["LULUCF"], axis=1).sum(axis=1)
co2_totals_excl_lulucf

Country_code
AT      62.124701
BE     120.309343
CH      44.534441
CZ     166.068823
DE    1052.348544
DK      54.685986
FR     400.777042
IT     438.008802
LU      11.847635
NL     163.302301
PL     376.546518
SE      57.348825
dtype: float64

In [20]:
# Total emissions incl. LULUCF
co2_totals_incl_lulucf = co2_totals.sum(axis=1)
co2_totals_incl_lulucf

Country_code
AT      49.968104
BE     117.060288
CH      42.515838
CZ     160.298010
DE    1021.884333
DK      60.858777
FR     374.572263
IT     432.346810
LU      11.927560
NL     169.786966
PL     343.554029
SE      21.112932
dtype: float64

In [21]:
# Total emissions excl. LULUCF
co2_totals_excl_lulucf_2005 = co2_totals_2005.drop(["LULUCF"], axis=1).sum(axis=1)
co2_totals_excl_lulucf_2005

Country_code
AT     79.192432
BE    125.661688
CH     45.915746
CZ    126.738399
DE    866.388549
DK     52.164863
FR    426.412463
IT    500.005676
LU     12.105448
NL    177.897237
PL    323.161267
SE     53.852857
dtype: float64

In [22]:
# Total emissions incl. LULUCF 2005
co2_totals_incl_lulucf_2005 = co2_totals_2005.sum(axis=1)
co2_totals_incl_lulucf_2005

Country_code
AT     68.414161
BE    124.024742
CH     43.712103
CZ    119.316791
DE    851.168952
DK     57.805099
FR    377.355895
IT    463.962937
LU     11.462305
NL    183.428661
PL    271.748435
SE     19.936933
dtype: float64

In [23]:
co2_totals_2005_all = pd.read_csv("../data/co2_totals_2005.csv", index_col=0)
co2_totals_excl_lulucf_2005_all = co2_totals_2005_all.drop(["LULUCF"], axis=1).sum(axis=1)

In [27]:
co2_totals_2005_all = pd.read_csv("../data/co2_totals_2005.csv", index_col=0)
co2_totals_incl_lulucf_2005_all = co2_totals_2005_all.sum(axis=1)

In [28]:
emission_share_included_ctrys_excl_lulucf = co2_totals_excl_lulucf_2005.sum()/co2_totals_excl_lulucf_2005_all.sum()
emission_share_included_ctrys_excl_lulucf

0.6245472968134671

In [29]:
emission_share_included_ctrys_incl_lulucf = co2_totals_incl_lulucf_2005.sum()/co2_totals_incl_lulucf_2005_all.sum()
emission_share_included_ctrys_incl_lulucf

0.6327636937085288

### ESR targets 2030 (compared to 2005) [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en)

In [9]:
esr_targets = {
    "AT": 0.48,
    "BE": 0.47,
    "CZ": 0.26,
    "DE": 0.5,
    "DK": 0.5,
    "FR": 0.475,
    "IT": 0.437,
    "LU": 0.5,
    "NL": 0.48,
    "PL": 0.177,
    "SE": 0.5,
}

### Country specific emission reduction targets

| Country | 2030 | 2035 | 2040 | 2045 | 2050 | Source |
|---------|------|------|------|------|------|--------|
| DE | 65% below 1990 excl. LULUCF | - | 85% below 1990 excl. LULUCF | 100% | - | [Source](https://www.bundesregierung.de/breg-de/themen/klimaschutz/klimaschutzgesetz-2021-1913672) |
| AT | 48% below 2005 excl. LULUCF | - | 100% | - | - | [Source1](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) [Source2](https://www.sustainability.gov/pdfs/austria-nzgi-roadmap.pdf) |
| BE | 55% below 1990 excl. LULUCF | - | - | - | 100% | [Source](https://www.sustainability.gov/pdfs/belgium-nzgi-roadmap.pdf) |
| CH | 50% below 1990 excl. LULUCF | - | - |  - | 100% | [Source](https://www.bafu.admin.ch/bafu/en/home/topics/climate/info-specialists/emission-reduction/reduction-targets.html) |
| CZ | 26% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://tracker.carbongap.org/region/czech-republic/) |
| DK | 70% below 1990 incl. LULUCF and 50% below 2005 excl. LULUCF | - | - |  100% | 110% | [Source](https://unfccc.int/sites/default/files/resource/SBI60_MA_DNK-7June2024-Presentation.pdf) |
| FR | 47.5% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) |
| IT | 43.7% below 2005 excl. LULUCF | - | - |  - | 100% | [Source1](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) [Source2](https://www.iea.org/countries/italy#) |
| LU | 55% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://gouvernement.lu/en/dossiers/2023/2023-pnec.html) |
| NL | 55% below 1990 | - | - |  - | 100% | [Source](https://commission.europa.eu/document/download/b6d21e56-4297-4b91-a692-300716209f72_en?filename=NL_FINAL%20UPDATED%20NECP%202021-2030%20%28English%29.pdf) |
| PL | 17.7% below 2005 excl. LULUCF | - | - |  - |  | [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) |
| SE | 63% below 1990 | - | 75% below 1990 |  100% |  | [Source](https://www.bafu.admin.ch/bafu/en/home/topics/climate/info-specialists/emission-reduction/reduction-targets.html) |

In [10]:
# calculate linearly interpolated 2035 targets
# base 1990
target2035_base1990 = dict(DE = 0.75,
BE = 0.55 + (1-0.55)*(2035-2030)/(2050-2030),
CH = 0.625,
DK = 0.7 + (1-0.7)*(2035-2030)/(2045-2030),
NL = 0.55 + (1-0.55)*(2035-2030)/(2050-2030),
SE = 0.69,
PL = (esr_targets["PL"]/esr_targets["DE"])*0.75
)
# base 2005
# Polands 2035 target is deduced based on the ESR distrubution key in relation to the German interpolated target
target2035_base2005 = dict(AT = 0.74,
CZ = 0.26 + (1-0.26)*(2035-2030)/(2050-2030),
FR = 0.475 + (1-0.475)*(2035-2030)/(2050-2030),
IT = 0.437 + (1-0.437)*(2035-2030)/(2050-2030),
LU = 0.55 + (1-0.55)*(2035-2030)/(2050-2030)
)
print(target2035_base1990, target2035_base2005)

{'DE': 0.75, 'BE': 0.6625000000000001, 'CH': 0.625, 'DK': 0.7999999999999999, 'NL': 0.6625000000000001, 'SE': 0.69, 'PL': 0.26549999999999996} {'AT': 0.74, 'CZ': 0.445, 'FR': 0.60625, 'IT': 0.57775, 'LU': 0.6625000000000001}


In [11]:
target_2035 = pd.DataFrame(index=co2_totals.index)
target_2035["base1990"] = target2035_base1990
target_2035["base2005"] = target2035_base2005
target_2035

base1990  base2005
Country_code                    
AT                 NaN   0.74000
BE              0.6625       NaN
CH              0.6250       NaN
CZ                 NaN   0.44500
DE              0.7500       NaN
DK              0.8000       NaN
FR                 NaN   0.60625
IT                 NaN   0.57775
LU                 NaN   0.66250
NL              0.6625       NaN
PL              0.2655       NaN
SE              0.6900       NaN

In [12]:
# Calculate aggregated 1990 emissions in MtCO2
co2_1990_aggregated = co2_totals_excl_lulucf.sum()
co2_1990_aggregated

2947.9029606100003

In [13]:
# Calculate absolute target emissions in 2035 based on shares
target_2035["absolute_emissions"] = ((1-target_2035.base1990) * co2_totals_excl_lulucf).combine_first((1-target_2035.base2005) * co2_totals_excl_lulucf_2005)
target_2035_aggregated = target_2035.absolute_emissions.sum()
target_2035_aggregated

1154.8379684469976

In [14]:
# Calculate share of 1990 emissions
round(target_2035_aggregated/co2_1990_aggregated,3)

0.392

Alternate approach: For missing values take interpolated emission reduction target for Germany 2035 (75%). For other countries take distribution key from [EU effort sharing regulation](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en). Then calculate total emissions based on 1990 values and weighted emission reduction target.

In [15]:
target_2035

base1990  base2005  absolute_emissions
Country_code                                        
AT                 NaN   0.74000           20.590032
BE              0.6625       NaN           40.604403
CH              0.6250       NaN           16.700415
CZ                 NaN   0.44500           70.339812
DE              0.7500       NaN          263.087136
DK              0.8000       NaN           10.937197
FR                 NaN   0.60625          167.899907
IT                 NaN   0.57775          211.127397
LU                 NaN   0.66250            4.085589
NL              0.6625       NaN           55.114527
PL              0.2655       NaN          276.573417
SE              0.6900       NaN           17.778136

## Europe's 2040 climate target
Source: https://eur-lex.europa.eu/resource.html?uri=cellar:6c154426-c5a6-11ee-95d9-01aa75ed71a1.0001.02/DOC_3&format=PDF

In [3]:
# calculate linearly interpolated 2035 targets
target_sector = pd.read_csv("../data/co2_total_sector.csv", index_col=0)
target_sector['2035 S2'] = (target_sector['2030'] + target_sector['2040 S2'])/2
target_sector_share = target_sector/target_sector.loc['Total Gross GHG Emissions',:]

In [4]:
target_sector

2005  2015  2030  2040 S1  2040 S2  2040 S3  \
MtC02-eq                                                                   
Total Gross GHG Emissions    4641  3914  2301     1273      943      748   
Power and district heating   1300  1012   339      123       42       23   
Other Energy sectors*         277   237   133       71       59       53   
Industry (Energy)             469   360   232      126       94       75   
Domestic Transport            822   772   583      190      143      120   
Residential and Services**    648   514   221      119       92       75   
Industry (Non-Energy)         343   233   157      139       88       14   
Other Non-Energv sectors***   101   130    56       33       26       25   
Intra-EU aviation              35    38    43       31       29       28   
Intra-EU navigation            31    27    25        7        6        4   
50% extra-EU maritime MRV      50    42    44       14       11        9   
Agriculture****               390   385   361      351      302      271   
Waste                         155   118    87       68       55       55   
C02 calibration                15    43    24        3       -1       -1   
Non-C02 calibration             5     2    -3       -3       -3       -3   

                             2040 LIFE  2050 S1  2050 S2  2050 S3  2050 LIFE  \
MtC02-eq                                                                       
Total Gross GHG Emissions          740      416      413      411     360.00   
Power and district heating          34       21       22       19      15.00   
Other Energy sectors*               57       39       39       38      36.00   
Industry (Energy)                   86        6        6        9      11.00   
Domestic Transport                 134       10        8        7       9.00   
Residential and Services**          92       20       19       19      29.00   
Industry (Non-Energy)               13        7        7        7       7.00   
Other Non-Energv sectors***         25       23       22       22      22.00   
Intra-EU aviation                   14       14       12       11      10.00   
Intra-EU navigation                  0        0        0        0       0.00   
50% extra-EU maritime MRV            0        0        0        0       0.00   
Agriculture****                    209      249      249      249     194.00   
Waste                               55       32       32       32       0.32   
C02 calibration                     -1        0        0        0       0.00   
Non-C02 calibration                 -3       -3       -3       -3      -3.00   

                             2035 S2  
MtC02-eq                              
Total Gross GHG Emissions     1622.0  
Power and district heating     190.5  
Other Energy sectors*           96.0  
Industry (Energy)              163.0  
Domestic Transport             363.0  
Residential and Services**     156.5  
Industry (Non-Energy)          122.5  
Other Non-Energv sectors***     41.0  
Intra-EU aviation               36.0  
Intra-EU navigation             15.5  
50% extra-EU maritime MRV       27.5  
Agriculture****                331.5  
Waste                           71.0  
C02 calibration                 11.5  
Non-C02 calibration             -3.0

In [6]:
# Select the included sector in the model (other energy sectors include non-CO2 GHG emissions and are thus excluded)
include_sector = ['Power and district heating','Residential and Services**'] #,'Other Energy sectors*']

target_sector_share_2035 = target_sector_share.loc[include_sector,'2035 S2'].sum()
target_sector_share_2035

0.27311960542540076

In [33]:
target_2035['included_sector_emissions'] = target_2035['absolute_emissions'] * target_sector_share_2035
target_2035

base1990  base2005  absolute_emissions  \
Country_code                                           
AT                 NaN   0.74000           20.590032   
BE              0.6625       NaN           40.604403   
CH              0.6250       NaN           16.700415   
CZ                 NaN   0.44500           70.339812   
DE              0.7500       NaN          263.087136   
DK              0.8000       NaN           10.937197   
FR                 NaN   0.60625          167.899907   
IT                 NaN   0.57775          211.127397   
LU                 NaN   0.66250            4.085589   
NL              0.6625       NaN           55.114527   
PL              0.2655       NaN          276.573417   
SE              0.6900       NaN           17.778136   

              included_sector_emissions  
Country_code                             
AT                             4.404896  
BE                             8.686639  
CH                             3.572777  
CZ                            15.048036  
DE                            56.283130  
DK                             2.339832  
FR                            35.919401  
IT                            45.167205  
LU                             0.874044  
NL                            11.790839  
PL                            59.168296  
SE                             3.803337

In [34]:
# Calculate share of 1990 emissions
round(target_2035['included_sector_emissions'].sum()/co2_1990_aggregated,3)

0.084